var/spack/repos/builtin/packages/py-llvmlite/package.py

```diff
@@ -21,7 +21,8 @@ class PyLlvmlite(PythonPackage):
     depends_on('py-setuptools', type='build')
     depends_on('python@2.6:2.8,3.4:', type=('build', 'run'))
     depends_on('py-enum34', type=('build', 'run'), when='^python@:3.3.99')
-    depends_on('llvm@7.0:7.99', when='@0.27.0:')
+    depends_on('llvm@7.0:8.0.99', when='@0.29.0:')
+    depends_on('llvm@7.0:7.99', when='@0.27.0:0.28.99')
     depends_on('llvm@6.0:6.99', when='@0.23.0:0.26.99')
     depends_on('llvm@4.0:4.99', when='@0.17.0:0.20.99')
```

Comment:
Might as well add the whole compatibility table if you want:
```rst
=================  ========================
llvmlite versions  compatible LLVM versions
=================  ========================
0.29.0 - ...       7.0.x, 7.1.x, 8.0.x
0.27.0 - 0.28.0    7.0.x
0.23.0 - 0.26.0    6.0.x
0.21.0 - 0.22.0    5.0.x
0.17.0 - 0.20.0    4.0.x
0.16.0 - 0.17.0    3.9.x
0.13.0 - 0.15.0    3.8.x
0.9.0 - 0.12.1     3.7.x
0.6.0 - 0.8.0      3.6.x
0.1.0 - 0.5.1      3.5.x
=================  ========================
```
Also add a comment saying this compatibility table is in `README.rst`

Reply:
Since there are no versions below 0.17 in spack, I don't think it's necessary to add their dependencies.